### PR TITLE
Load surveys from the filesystem, compile, + list over HTTP

### PR DESCRIPTION
**Pull request by mojodna**
Reads a pile of files (a copy of iD presets modified to fit needs), compiles them into a bundle (tarball), and stores them in the user data directory. The list of surveys is propagated to the renderer thread using a bridged `dispatch` method. Surveys can also be listed over HTTP.

Fixes osmlab/field-data-collection#102
Fixes osmlab/field-data-collection#87
Refs osmlab/field-data-collection#74

### CODE REVIEW

File: lib/surveys.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
+'import strict'
```
Comment (dereklieu) [inline lib/surveys.js]: Typo :)

File: main.js [hunk 2]
```diff
@@ -38,11 +60,15 @@ function setupMenu () {
...
+
+  // TODO see Darwin-specific menu configuration here for better Windows
+  // behavior:
+  // https://github.com/electron/electron/blob/master/docs/api/menu.md#main-process
 }
 
-const userDataPath = path.join(app.getPath('userData'), 'org.osm-labs.field-data-coordinator')
-mkdirp.sync(userDataPath)
-const dbPath = path.join(userDataPath, 'db')
+function setupFileIPCs (main, inChannel, outChannel) {}
```
Comment (dereklieu) [inline main.js]: I just cribbed this from react-mapfilter desktop, it doesn't need to be here.

File: src/index.jsx [hunk 3]
```diff
@@ -21,16 +24,22 @@ const store = createStore(reducers, applyMiddleware(sagaMiddleware))
 sagaMiddleware.run(rootSaga)
 store.dispatch(sync())
 
-ReactDOM.render((
+// register a dispatch listener so that actions can be sent between processes
+ipcRenderer.on('dispatch', (evt, payload) => store.dispatch(payload))
+ipcRenderer.send('redux-initialized')
```
Comment (dereklieu) [inline src/index.jsx]: 👍

File: lib/surveys.js (hunk 1 above)
Comment (mojodna) [inline lib/surveys.js]: JS dialects.

**Pull request merged**